agent/util-scripts/pbench-tool-meister-start

```diff
@@ -435,6 +456,14 @@ class ToolDataSink(BaseServer):
                     else:
                         time.sleep(0.1)
 
+        # TDS daemonization should create a PID file in the current working
+        # directory; confirm it exists, and record the path for `kill`.
+        pid_file = Path("pbench-tool-data-sink.pid").resolve()
+        if pid_file.exists():
```

Comment:
Are we sure the pid file will be created by the time the Tool Data Sink starts listening on the logging channel?

Also, the only time that pid file is created is when the Tool Data Sink is orchestrated by us.  Should this be conditionalized?

Reply:
This method is only called when we're orchestrating; the `pbench-tool-data-sink` command is defined here, with the `daemonize` argument set to `yes`; so even if we allowed someone else to call this code (which would be problematic), we know it'll be daemonized. The pidfile creation is done by the `DaemonContext` call in `tool_data_sink.py`; and while it's not instantaneous, we know that it'll be done by the time the process gets far enough to open the Redis channel (it's part of the `DaemonContext` `__enter__` due to the symbiotic reference to the `pidfile.PIDFile` object we give it.

So `pbench-tool-meister-start` only gets here if we own the orchestration; and we'll only push the `kill` cleanup action when we're orchestrating.

The linkage here isn't ideal, since both sides are assuming the `pbench-tool-data-sink.pid` name. I briefly considered adding the pid file name as an extra command line parameter, but decided it wasn't worthwhile at this point.

Reply:
Thanks for confirming this is sufficient for now.